fastly/tls_mutual_authentication.go

```diff
@@ -0,0 +1,212 @@
+package fastly
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/google/jsonapi"
+)
+
+// TLSMutualAuthentication represents a mutual authentication certificate.
+type TLSMutualAuthentication struct {
+	Activations []*TLSActivation `jsonapi:"relation,tls_activations"`
+	CreatedAt   *time.Time       `jsonapi:"attr,created_at,iso8601"`
+	Enforced    bool             `jsonapi:"attr,enforced"`
+	ID          string           `jsonapi:"primary,mutual_authentication"`
+	Name        string           `jsonapi:"attr,name"`
+	UpdatedAt   *time.Time       `jsonapi:"attr,updated_at,iso8601"`
+}
+
+// ListTLSMutualAuthenticationsInput is used as input to the Client.ListTLSMutualAuthentication function.
+type ListTLSMutualAuthenticationsInput struct {
+	// Include is a comma-separated list of related objects to include (optional).
+	Include string
+	// PageNumber is the required page index for pagination.
+	PageNumber int
+	// PageSize is the number of records per page.
+	PageSize int
+}
+
+// formatFilters converts user input into query parameters for filtering.
+func (i *ListTLSMutualAuthenticationsInput) formatFilters() map[string]string {
+	result := map[string]string{}
+	pairings := map[string]interface{}{
+		"include":      i.Include,
+		"page[size]":   i.PageSize,
+		"page[number]": i.PageNumber,
+	}
+
+	for key, value := range pairings {
+		switch t := reflect.TypeOf(value).String(); t {
+		case "string":
+			if value != "" {
+				result[key] = value.(string)
+			}
+		case "int":
+			if value != 0 {
+				result[key] = strconv.Itoa(value.(int))
+			}
+		}
+	}
+
+	return result
```

Comment:
Since there's only 3 things here, it's fewer lines of code to check them individually and avoid use of `reflect`:

```go
result := map[string]string{}
if i.Include != "" {
	result["include"] = i.Include
}
if i.PageSize > 0 {
	result["page[size]"] = strconv.Itoa(i.PageSize)
}
if i.PageNumber > 0 {
	result["page[number]"] = strconv.Itoa(i.PageNumber)
}
return result
```